src/app/api/routes/devices.py

```diff
@@ -76,11 +79,21 @@ async def get_my_device(me: DeviceOut = Security(get_current_device, scopes=["de
 
 
 @router.get("/", response_model=List[DeviceOut], summary="Get the list of all devices")
-async def fetch_devices(_=Security(get_current_access, scopes=[AccessType.admin])):
+async def fetch_devices(requester=Security(get_current_access, scopes=[AccessType.admin, AccessType.user]),
+                        session=Depends(get_session)):
     """
     Retrieves the list of all devices and their information
     """
-    return await crud.fetch_all(devices)
+    if await is_admin_access(requester.id):
+        return await crud.fetch_all(devices)
+    else:
+        print(session.query(models.Devices).first().access)
+        retrieved_devices = (session.query(models.Devices)
+                             .join(models.Accesses)
+                             .filter(models.Accesses.group_id == requester.group_id).all())
```

Comment:
same comment here

Reply:
same reply as in the other comment, only the devices columns are retrieved; and doing a single SQL query can't be beaten.